### PR TITLE
Azure VirtualMachine should Implement GyroInstance

### LINKED ISSUE #55: Azure VirtualMachine should Implement GyroInstance
Virtual Machines need to implement the` GyroInstance` interface in order for certain functionality in the SSH plugin to work in Azure. This includes list functionality and getting IP addresses.

More functionality can be written that is cloud agnostic that should use GyroInstance as the medium. Given that we would want `VirtualMachine` to be included in this functionality, it should implement `GyroInstance`.

### PR DESCRIPTION
**Pull request by deepanjan90**
Fixes #55

### CODE REVIEW

File: src/main/java/gyro/azure/compute/VirtualMachineResource.java [hunk 1]
```diff
@@ -807,6 +885,7 @@ public void create(GyroUI ui, State state) {
 
         setId(virtualMachine.id());
         setVmId(virtualMachine.vmId());
+        copyFrom(virtualMachine);
```
Comment (tloisel1) [inline src/main/java/gyro/azure/compute/VirtualMachineResource.java]: Seems like the use of `copyFrom()` is a doing more than we expect. I understand that some of the fields cannot be known until the VM is actually launched. Will running this extra logic have any unintentional side effects?

File: src/main/java/gyro/azure/compute/VirtualMachineResource.java [hunk 2]
```diff
@@ -831,7 +910,8 @@ public void update(GyroUI ui, State state, Resource current, Set<String> changed
             }
         }
 
-        update.apply();
+        virtualMachine = update.apply();
+        copyFrom(virtualMachine);
```
Comment (tloisel1) [inline src/main/java/gyro/azure/compute/VirtualMachineResource.java]: See above

File: src/main/java/gyro/azure/compute/VirtualMachineResource.java [hunk 3]
```diff
@@ -848,4 +928,44 @@ private String getEncodedCustomData() {
     private String getDecodedCustomData(String data) {
         return !ObjectUtils.isBlank(data) ? new String(Base64.getDecoder().decode(data.getBytes())) : null;
     }
+
```
Comment (tloisel1) [inline src/main/java/gyro/azure/compute/VirtualMachineResource.java]: Please add a comment line that clearly indicates the following methods are for `GyroInstance` Support.

File: src/main/java/gyro/azure/compute/VirtualMachineResource.java (hunk 1 above)
Comment (deepanjan90) [reply src/main/java/gyro/azure/compute/VirtualMachineResource.java]: The call to copy from is especially needed in this case as the output fields (which are not configured by the user) are not saved in the state until we set it explicitly using the setter, after (create/update). The fields like `launchDate` and `state` would not be present when the ssh plugin calls the `gyro ssh` or `gyro list`

Comment (tloisel1) [reply src/main/java/gyro/azure/compute/VirtualMachineResource.java]: Ok, my concern was that calling copyFrom() would have performance implications.

**Pull request merged**